src/Microsoft.HttpRepl.IntegrationTests/Preferences/PreferencesTests.cs

```diff
@@ -0,0 +1,97 @@
+using System.Collections.Generic;
+using Microsoft.HttpRepl.IntegrationTests.Mocks;
+using Microsoft.HttpRepl.Preferences;
+using Microsoft.HttpRepl.UserProfile;
+using Xunit;
+using Prefs = Microsoft.HttpRepl.Preferences.Preferences;
+
+namespace Microsoft.HttpRepl.IntegrationTests.Preferences
+{
+    public class PreferencesTests
+    {
+        [Fact]
+        public void ReadPreferences_NoPreferencesFile_AllDefaults()
+        {
+            SetupPreferences(out Prefs preferences, out _);
+
+            ConfirmAllPreferencesAreDefaults(preferences);
+        }
+
+        [Fact]
+        public void ReadPreferences_BlankPreferencesFile_AllDefaults()
+        {
+            SetupPreferences(out Prefs preferences, out MockedFileSystem fileSystem);
+            fileSystem.AddFile(preferences.PreferencesFilePath, "");
+
+            ConfirmAllPreferencesAreDefaults(preferences);
+        }
+
+        [Fact]
+        public void ReadPreferences_InvalidPreferencesFile_AllDefaults()
+        {
+            SetupPreferences(out Prefs preferences, out MockedFileSystem fileSystem);
+            fileSystem.AddFile(preferences.PreferencesFilePath, "This is not a valid preferences file.");
+
+            ConfirmAllPreferencesAreDefaults(preferences);
+        }
+
+        [Fact]
+        public void ReadPreferences_PartiallyInvalidPreferencesFile_ValidPrefsAreSet()
+        {
+            SetupPreferences(out Prefs preferences, out MockedFileSystem fileSystem);
+            string settingName = WellKnownPreference.DefaultEditorCommand;
+            string expectedValue = "Code.exe";
+            string prefsFileContent = $@"This first line is invalid for a prefs file.
+{settingName}={expectedValue}
+This third line is invalid as well";
+            fileSystem.AddFile(preferences.PreferencesFilePath, prefsFileContent);
+
+            Dictionary<string, string> preferencesDictionary = preferences.ReadPreferences(HttpState.CreateDefaultPreferences());
+
+            Assert.Equal(expectedValue, preferencesDictionary[settingName]);
+        }
+
+        [Fact]
+        public void WritePreferences_SomeDefault_OnlyWritesNonDefaultValues()
```

Comment:
it feels like there's a second test lingering in the OnlyWritesNonDefaultValues shadow:

1. load prefs from a file that contains a non-default setting
2. Set that pref to the default value
3. Write preferences
4. Verify that preference is removed from the file?

Reply:
Good call. I added a test for that.